src/grpc/dp_grpc_impl.c

```diff
@@ -462,7 +453,7 @@ static int dp_process_delete_prefix(struct dp_grpc_responder *responder)
 	target_vnf.vni = port->iface.vni;
 
 	dp_assign_ip_address(&target_vnf.alias_pfx.ol, &request->addr);
```

Comment:
Is this needed as an exported thing now? I think you can just assign it here and then do the check in the "key" creation.

I think it would be better because the caller cannot forget to use this special assignment, otherwise stuff will break.

Reply:
after doubling checking, this function is exported due to some leftovers. One further step is made to integrate this function into `dp_fill_vnf_data`, as it makes most of the juice from two and reduces code complexity. Unless `dp_assign_ip_address` is needed in other places, the changes now should resolve our concerns so far.